File: sample/exp32/exp32.clj
(defrule plus-assign @{:kind "Binop" 
                       :op "=" 
                       :LHS {:kind "DRE" 
                             :type [{:kind "Pointer-type" 
                                     :Pointee {:kind "Pointer-type" 
                                               :Pointee {:kind "Int-type" :volatile "true"}}}]} 
                       :RHS {:kind "Unop" 
                             :type [{:kind "Pointer-type" 
                                     :Pointee {:kind "Pointer-type" :Pointee {:kind "Int-type"}}} 
                                    {:kind "Pointer-type" 
                                     :Pointee {:kind "Pointer-type" :Pointee {:kind "Int-type"}}} 
                                    {:kind "Pointer-type" :Pointee {:kind "Pointer-type"}}]
                             }} 
  `("Very bad!!! exp32") )

File: sample/exp32/exp32.c
main(){
  static volatile int **ipp;
  static int *ip;
  static volatile int i = 0;
  int **x;
  
  printf("i = %d.\n", i);

  ipp = &ip;
  x = (int**) &ip;
  ipp = x;
  *ipp = &i;
  if (*ip != 0) {
  }
}
